src/main/java/edu/mit/puzzle/cube/core/model/User.java

```diff
@@ -0,0 +1,26 @@
+package edu.mit.puzzle.cube.core.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+
+import java.util.List;
+
+@AutoValue
+@JsonDeserialize(builder = AutoValue_User.Builder.class)
+public abstract class User {
+    @AutoValue.Builder
+    public static abstract class Builder {
+        @JsonProperty("username") public abstract Builder setUsername(String username);
+        @JsonProperty("permissions") public abstract Builder setPermissions(List<String> permissions);
+
+        public abstract User build();
+    }
+
+    public static Builder builder() {
+        return new AutoValue_User.Builder();
+    }
+
+    @JsonProperty("username") public abstract String getUsername();
+    @JsonProperty("permissions") public abstract List<String> getPermissions();
```

Comment:
This is going to be a problem, or at least a confusing inconsistency, if we assign permissions to roles rather than directly to users, which I think that we should due to having individual users at least on the game control team - and possibly standard and admin accounts for solving teams?

Shiro doesn't appear to have a Subject.listPermissions() or equivalent, which I totally get because that is HARD, but it means that we'll have a hard time populating this field even on a Role object.

Basically, I'm not sure what we would actually do with this in the model object, not sure we can make it actually accurate, and not sure it's in the right place.